src/isolate.rs

```diff
@@ -1030,3 +1042,128 @@ where
     f.to_c_fn()
   }
 }
+
+/// A special hasher that is optimized for hashing `std::any::TypeId` values.
+/// It can't be used for anything else.
+#[derive(Clone, Default)]
+pub(crate) struct TypeIdHasher {
+  state: Option<u64>,
+}
+
+impl Hasher for TypeIdHasher {
+  fn write(&mut self, _bytes: &[u8]) {
+    panic!("TypeIdHasher::write() called unexpectedly");
+  }
+
+  #[inline]
+  fn write_u64(&mut self, value: u64) {
+    // `TypeId` values are actually 64-bit values which themselves come out of
+    // some hash function, so it's unnecessary to shuffle their bits further.
+    assert_eq!(size_of::<TypeId>(), size_of::<u64>());
+    assert_eq!(align_of::<TypeId>(), size_of::<u64>());
```

Comment:
```rust

const _: () = {
   // `TypeId` values are actually 64-bit values which themselves come out of
   // some hash function, so it's unnecessary to shuffle their bits further.
   assert!(size_of::<TypeId>() == size_of::<u64>());
   assert!(align_of::<TypeId>() == size_of::<u64>());
};
```
This can be a constant assert.